src/CVRF/cvrf_eval.c

```diff
@@ -0,0 +1,423 @@
+	oval_object_add_object_content(object, object_content);
+
+	oscap_free(object_id);
+	return object;
+}
+
+static struct oval_state *get_new_oval_state_for_cvrf(struct oval_definition_model *def_model,
+		struct cvrf_rpm_attributes *attributes, int stateNo) {
+
+	char *state_id = get_oval_id_string("ste", stateNo);
+
+	// Entity (Package name match)
+	struct oval_entity *state_entity = oval_entity_new(def_model);
+	oval_entity_set_name(state_entity, "name");
+	oval_entity_set_operation(state_entity, OVAL_OPERATION_PATTERN_MATCH);
+	struct oval_value *state_value = oval_value_new(OVAL_DATATYPE_STRING, attributes->rpm_name);
+	oval_entity_set_value(state_entity, state_value);
+	// Content (Package name match)
+	struct oval_state_content *state_content = oval_state_content_new(def_model);
+	oval_state_content_set_entity(state_content, state_entity);
+
+
+	// Entity (EVR format less than)
+	struct oval_entity *evr_entity = oval_entity_new(def_model);
+	oval_entity_set_name(evr_entity, "evr");
+	oval_entity_set_datatype(evr_entity, OVAL_DATATYPE_EVR_STRING);
+	oval_entity_set_operation(evr_entity, OVAL_OPERATION_LESS_THAN);
+	struct oval_value *evr_value = oval_value_new(OVAL_DATATYPE_EVR_STRING, attributes->evr_format);
+	oval_entity_set_value(evr_entity, evr_value);
+	// Content (EVR format less than)
+	struct oval_state_content *evr_content = oval_state_content_new(def_model);
+	oval_state_content_set_entity(evr_content, evr_entity);
+
+	struct oval_state *state = oval_definition_model_get_new_state(def_model, state_id);
+	oval_state_set_comment(state, attributes->full_package_name);
+	oval_state_set_subtype(state, OVAL_LINUX_RPM_INFO);
+	oval_state_set_operator(state, OVAL_OPERATOR_AND);
+	oval_state_set_version(state, 1);
+	oval_state_add_content(state, state_content);
+	oval_state_add_content(state, evr_content);
+
+	//oscap_free(rpm_name_match);
+	oscap_free(state_id);
+	return state;
+}
+
+static struct oval_test *get_new_rpminfo_test_for_cvrf(struct oval_definition_model *def_model,
+		struct cvrf_rpm_attributes *attributes, int testNo) {
+
+	char *test_id = get_oval_id_string("tst", testNo);
+	struct oval_test *rpm_test = oval_test_new(def_model, test_id);
+	oval_test_set_subtype(rpm_test,OVAL_LINUX_RPM_INFO);
+	oval_test_set_version(rpm_test, 1);
+	oval_test_set_check(rpm_test, OVAL_CHECK_AT_LEAST_ONE);
+	oval_test_set_existence(rpm_test, OVAL_AT_LEAST_ONE_EXISTS);
+
+	oval_test_add_state(rpm_test, get_new_oval_state_for_cvrf(def_model, attributes, testNo));
+	oval_test_set_object(rpm_test, get_new_oval_object_for_cvrf(def_model, attributes, testNo));
+
+	return rpm_test;
+}
+
+static struct oval_definition *get_new_oval_definition_for_cvrf(struct oval_definition_model *def_model,
+		struct cvrf_rpm_attributes *attributes, int index) {
+
+	char *definition_id = get_oval_id_string("def", index);
+	struct oval_definition *definition = oval_definition_model_get_new_definition(def_model, definition_id);
```

Comment:
free definition_id here. generally, free things as soon as possible but not sooner.